### PR TITLE
windows: fix build error on MinGW

### PR DESCRIPTION
**Pull request by kenhys**
Patch via
https://github.com/msys2/MINGW-packages/commit/9c76ae395dd490b259d270608b47d130c6245490#diff-b266d04f16a509dc5bd3a3d12ca9df53

### CODE REVIEW

File: vendor/mruby/Makefile.am [hunk 1]
```diff
@@ -8,6 +8,7 @@ EXTRA_DIST =					\
 	mruby-dir/src/Win/dirent.c
 
 DEFAULT_INCLUDES =						\
+	-I$(top_builddir)			\
```
Comment (kou) [inline vendor/mruby/Makefile.am]: The trailing `\` isn't aligned.d

Comment (kenhys) [reply vendor/mruby/Makefile.am]: Thanks fixed.

**Comment by kenhys** [comment]
Yes. (config.h not found)
Here is the actual error message:

```
In function 'parser_yylex',
    inlined from 'yylex' at /work/groonga/groonga.clean/vendor/mruby-source/mrbgems/mruby-compiler/core/parse.y:5979:10,
    inlined from 'yyparse' at /work/groonga/groonga.clean/vendor/mruby-build/host/mrbgems/mruby-compiler/core/y.tab.c:5442:16:
/work/groonga/groonga.clean/vendor/mruby-source/mrbgems/mruby-compiler/core/parse.y:5012:11: warning: 'strncat' output truncated before terminating nul copying 1 byte from a string of the same length [-Wstringop-truncation]
  CC       mruby-env/src/env.lo
  CC       mruby-file-stat/src/file-stat.lo
../../../groonga-9.0.4/vendor/mruby/mruby-file-stat/src/file-stat.c:11:10: fatal error: config.h: No such file or directory
   11 | #include "config.h"
      |          ^~~~~~~~~~
compilation terminated.
  CC       mruby-dir/src/dir.lo
make[4]: *** [Makefile:1269: mruby-file-stat/src/file-stat.lo] Error 1
```
[groonga-no-patch.log](https://github.com/groonga/groonga/files/3366811/groonga-no-patch.log)

**Comment by kou** [comment]
mruby-file-stat should use `config.h` generated by mruby-file-stat.
It should not use Groonga's `config.h`.

**Comment by kenhys** [comment]
So, this build error shoulb be fixed in mruby-file-stat side.

**Comment by kou** [comment]
We already have code for this case such as https://github.com/groonga/groonga/blob/master/vendor/mruby/Makefile.am#L43 .
I'm not sure why this is caused.

**Comment by kenhys** [comment]
It seems that 
mingw-w64-groonga\src\build-x86_64\vendor\mruby\mruby-file-stat\src\config.h exists.

According to verbose build log, under src/groonga-9.0.4 is expected?
```
libtool: compile:  x86_64-w64-mingw32-gcc -DHAVE_CONFIG_H -I../../../groonga-9.0.4/vendor/mruby/../mruby-source/include -I../../../groonga-9.0.4/vendor/mruby/../mruby-source/src -I../../../groonga-9.0.4/vendor/mruby/../mruby-source/mrbgems/mruby-compiler/core -I../../../groonga-9.0.4/vendor/mruby/../mruby-source/mrbgems/mruby-io/include -I../../../groonga-9.0.4/vendor/mruby/../mruby-source/mrbgems/mruby-time/include -I../../../groonga-9.0.4/vendor/mruby/../onigmo-source -DHAVE_ONIGMO_H -DMRB_INT64 -DMRB_BUILD_AS_DLL -DMRB_CORE -D_FORTIFY_SOURCE=2 -D__USE_MINGW_ANSI_STDIO=1 -march=x86-64 -mtune=generic -O2 -pipe -Wall -Wno-unused-but-set-variable -Wno-pointer-sign -Wno-declaration-after-statement -Wformat -Wno-format-truncation -Wstrict-aliasing=2 -Wdisabled-optimization -Wfloat-equal -Wpointer-arith -Wbad-function-cast -Wwrite-strings -Wmissing-field-initializers -Wno-float-equal -Wno-bad-function-cast -MT mruby-file-stat/src/file-stat.lo -MD -MP -MF mruby-file-stat/src/.deps/file-stat.Tpo -c ../../../groonga-9.0.4/vendor/mruby/mruby-file-stat/src/file-stat.c  -DDLL_EXPORT -DPIC -o mruby-file-stat/src/.libs/file-stat.o
../../../groonga-9.0.4/vendor/mruby/mruby-file-stat/src/file-stat.c:11:10: fatal error: config.h: No such file or directory
 #include "config.h"
          ^~~~~~~~~~
compilation terminated.
```

**Comment by kou** [comment]
> under src/groonga-9.0.4 is expected?

Why?

`-I$(builddir)/mruby-file-stat/src` will be needed.

**Comment by kenhys** [comment]
> -I$(builddir)/mruby-file-stat/src will be needed.

This is what actually needed to fix(verified). I did wrong comment in  https://github.com/groonga/groonga/pull/1003#issuecomment-509100943

File: vendor/mruby/Makefile.am [hunk 2]
```diff
@@ -12,7 +12,8 @@ DEFAULT_INCLUDES =						\
 	-I$(srcdir)/../mruby-source/src				\
 	-I$(srcdir)/../mruby-source/mrbgems/mruby-compiler/core	\
 	-I$(srcdir)/../mruby-source/mrbgems/mruby-io/include	\
-	-I$(srcdir)/../mruby-source/mrbgems/mruby-time/include
+	-I$(srcdir)/../mruby-source/mrbgems/mruby-time/include	\
+	-I$(builddir)/mruby-file-stat/src
```
Comment (kou) [inline vendor/mruby/Makefile.am]: `$(builddir)/...` should be listed before `$(srcdir)/...`.

**Comment by kou** [comment]
I think that we need a CI entry for MinGW on Windows for this case.
(It should be done in another pull request.)

File: vendor/mruby/Makefile.am (hunk 2 above)
Comment (kenhys) [reply vendor/mruby/Makefile.am]: I've fixed. Thanks!

**Comment by kenhys** [comment]
@kou Thanks! I've fixed listed order.

**Comment by kenhys** [comment]
>I think that we need a CI entry for MinGW on Windows for this case.
(It should be done in another pull request.)

I'll try it.

**Pull request merged**